src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class type) {
+
+        switch (type.getSimpleName()) {
+            case "Truck":
+                return new Truck().get();
+            case "Bulldozer":
+                return new Bulldozer().get();
+            case "Excavator":
+                return new Excavator().get();
```

Comment:
The use of the `get` method from `Truck`, `Bulldozer`, and `Excavator` classes is not ideal. These methods should not be part of these classes as they do not align with typical object-oriented design principles. Consider refactoring these classes to remove the `get` method and handle instance creation differently, possibly through a factory or producer pattern.